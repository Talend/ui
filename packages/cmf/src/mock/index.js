/**
 * The mock is in the src folder and there is reason for that
 * In the past without this we have added mocked states to other CMF libraries
 * and when we have let the state structure evolved, tests on those libs were still passing.
 * We believe they embed something that they are not responsible for.
 *
 * So mock are in the src folder because they are part of the library.
 * But they are not referenced in the index.js so they will not be in your build.
 * You have to import the mock from the complete path, ie:
 *
 * ```import mock from 'react-cmf/lib/mock';```
 */
import store from './store';
import Provider from './provider';

export { Provider, store };
