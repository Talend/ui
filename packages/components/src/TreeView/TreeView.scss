// TreeView styles

.tc-treeview {
	.tc-treeview-header {
		display: flex;
		justify-content: space-between;
		align-items: center;

		color: $brand-primary;
		line-height: 6rem;
		font-weight: bold;
		text-transform: uppercase;

		padding-left: $padding-large;
		border-bottom: 1px rgba(0, 0, 0, 0.12) solid;

		:global(.btn-link .tc-svg-icon) {
			stroke: $brand-primary;
		}
	}

	.tc-treeview-list {
		padding: $padding-small 0;
		overflow: auto;
		list-style-type: none;
	}
}
