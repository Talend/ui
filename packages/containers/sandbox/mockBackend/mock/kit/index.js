/* eslint-disable global-require */
module.exports = {
	titleMap: require('./titleMap.json'),
	validation: require('./validation.json'),
};
