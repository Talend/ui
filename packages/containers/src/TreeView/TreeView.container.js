import React from 'react';
import PropTypes from 'prop-types';
import { cmfConnect } from '@talend/react-cmf';
import { TreeView as Component } from '@talend/react-components';
import ImmutablePropTypes from 'react-immutable-proptypes';
import Immutable from 'immutable';
import omit from 'lodash/omit';

const OPENED_ATTR = 'opened';
const SELECTED_ATTR = 'selectedId';
export const DISPLAY_NAME = 'Container(TreeView)';
export const DEFAULT_PROPS = {
	idAttr: 'id',
	nameAttr: 'name',
	childrenAttr: 'children',
};
export const DEFAULT_STATE = new Immutable.Map({
	[OPENED_ATTR]: new Immutable.List(),
	[SELECTED_ATTR]: undefined,
});

function toggleState(prevProps, id) {
	const opened = prevProps.state.get(OPENED_ATTR);
	const index = opened.indexOf(id);
	if (index !== -1) {
		return prevProps.state.set(OPENED_ATTR, opened.delete(index));
	}
	return prevProps.state.set(OPENED_ATTR, prevProps.state.get(OPENED_ATTR).push(id));
}

function selectWrapper(prevProps, id) {
	const selected = prevProps.state.get(SELECTED_ATTR);
	if (id === selected) {
		return prevProps.state.set(SELECTED_ATTR, undefined);
	}
	return prevProps.state.set(SELECTED_ATTR, id);
}

/**
 * recursive function to apply change on all data and support attr mapping
 * @param {Array<Object>} items is the list of pure items, your data
 * @param {Object} props the configuration of the Tree container
 * @return {Array} of items ready to be put as the structure of TreeView component
 */
export function transform(items, props) {
	if (!items) {
		return undefined;
	}
	const state = props.state || DEFAULT_STATE;
	const selectedId = props[SELECTED_ATTR] || (state && state.get(SELECTED_ATTR));
	const opened = state && state.get(OPENED_ATTR).toJS();
	return items.map(item => ({
		...item,
		id: item[props.idAttr],
		selected: item[props.idAttr] === selectedId,
		toggled: item.toggled || opened.indexOf(item[props.idAttr]) !== -1,
		name: item[props.nameAttr],
		children: transform(item[props.childrenAttr], props),
	}));
}

/**
 * The TreeView React container
 */
class TreeView extends React.Component {
	static displayName = DISPLAY_NAME;
	static propTypes = {
		data: ImmutablePropTypes.List,
		idAttr: PropTypes.string,
		nameAttr: PropTypes.string,
		onClick: PropTypes.func,
		onSelect: PropTypes.func,
		onClickActionCreator: PropTypes.string,
		onSelectActionCreator: PropTypes.string,

		...cmfConnect.propTypes,
	};
	static defaultProps = DEFAULT_PROPS;

	constructor(props) {
		super(props);
		this.onSelect = this.onSelect.bind(this);
		this.onClick = this.onClick.bind(this);
	}

	onSelect(data) {
		this.props.setState(prevState => selectWrapper(prevState, data[this.props.idAttr]));
		if (this.props.onSelectActionCreator) {
			this.props.dispatchActionCreator(
				this.props.onSelectActionCreator,
				{
					type: 'select',
					source: 'TreeView',
					props: this.props,
				},
				data,
			);
		}
		if (this.props.onSelect) {
			this.props.onSelect(data);
		}
	}

	onClick(data) {
		this.props.setState(prevState => toggleState(prevState, data[this.props.idAttr]));
		if (this.props.onClickActionCreator) {
			this.props.dispatchActionCreator(
				this.props.onClickActionCreator,
				{
					type: 'toggle',
					source: 'TreeView',
					props: this.props,
				},
				data,
			);
		}
		if (this.props.onClick) {
			this.props.onClick(data);
		}
	}

	render() {
		if (!this.props.data) {
			return null;
		}
		const structure = transform(this.props.data.toJS(), this.props);
		const props = omit(this.props, cmfConnect.INJECTED_PROPS);
		return (
			<Component {...props} structure={structure} onSelect={this.onSelect} onClick={this.onClick} />
		);
	}
}

export function mapStateToProps(state, ownProps) {
	const props = {};
	if (ownProps.collection) {
		props.data = state.cmf.collections.getIn(ownProps.collection.split('.'));
		if (!props.data) {
			// eslint-disable-next-line no-console
			console.warn('TreeView.collection not found');
		}
	}
	return props;
}

export default cmfConnect({
	defaultState: DEFAULT_STATE,
	mapStateToProps,
})(TreeView);
