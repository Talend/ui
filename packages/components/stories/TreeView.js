/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { checkA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions';

import { TreeView, IconsProvider } from '../src/index';

const structure = [
	{ name: 'hitmonlee', children: [{ name: 'Hitmonchan' }], isOpened: false },
	{ name: 'pikachu', children: [{ name: 'raichu' }], isOpened: true },
	{
		id: 'selected',
		name: 'Abra',
		isOpened: true,
		children: [
			{
				name: 'Kadabra',
				isOpened: true,
				children: [
					{
						name: 'Alakazam',
					},
				],
			},
		],
	},
];

const structureWithIcons = [
	{
		name: 'hitmonlee',
		children: [{ name: 'Hitmonchan' }],
		isOpened: false,
		icon: {
			name: 'src-http://static.pokemonpets.com/images/monsters-images-300-300/106-Hitmonlee.png',
		},
	},
	{
		name: 'pikachu',
		children: [
			{
				name: 'raichu',
				icon: {
					name:
						'src-http://static.pokemonpets.com/images/monsters-images-300-300/2026-Shiny-Raichu.png',
				},
			},
		],
		isOpened: true,
		icon: {
			name:
				'src-http://static.pokemonpets.com/images/monsters-images-300-300/2025-Shiny-Pikachu.png',
		},
	},
	{
		id: 'selected',
		name: 'Abra',
		icon: { name: 'src-http://static.pokemonpets.com/images/monsters-images-300-300/63-Abra.png' },
		isOpened: true,
		children: [
			{
				name: 'Kadabra',
				icon: {
					name: 'src-http://static.pokemonpets.com/images/monsters-images-300-300/64-Kadabra.png',
				},
				isOpened: true,
				children: [
					{
						name: 'Alakazam',
						icon: {
							name:
								'src-http://static.pokemonpets.com/images/monsters-images-300-300/65-Alakazam.png',
						},
					},
				],
			},
		],
	},
];

const removeAction = [
	{
		action: action('itemRemoveCallback'),
		icon: 'talend-trash',
		label: 'remove element',
	},
];

const actions = [
	{
		action: action('itemAddCallback'),
		icon: 'talend-plus',
		label: 'Add Item',
	},
	{
		action: action('itemEditCallback'),
		icon: 'talend-pencil',
		label: 'Edit Item',
	},
	{
		action: action('itemRemoveCallback'),
		icon: 'talend-trash',
		label: 'Remove Item',
	},
];

const structureWithCounterAndAction = [
	{
		name: 'hitmonlee',
		isOpened: true,
		children: [
			{
				name: 'raichu',
				showCounter: true,
				counter: 111,
				actions: removeAction,
			},
		],
		counter: -1,
		showCounter: true,
		actions: removeAction,
	},
	{
		name: 'pikachu',
		isOpened: true,
		counter: 2911,
		showCounter: true,
		actions: removeAction,
	},
];

const structureWithActions = [
	{
		name: 'hitmonlee',
		isOpened: true,
		children: [
			{
				name: 'raichu',
				actions,
			},
		],
		actions,
	},
	{
		name: 'pikachu',
		isOpened: true,
		actions,
	},
];

const defaultProps = {
	id: 'my-treeview',
	structure,
	onSelect: action('onSelect'),
	onToggle: action('onToggle'),
	onToggleAllSiblings: action('onToggleAllSiblings'),
	selectedId: 'selected',
};

const withAddAction = {
	...defaultProps,
	addAction: action('added'),
};

const withHeader = {
	...withAddAction,
	headerText: 'some elements',
	addActionLabel: 'add element',
};

const withRemoval = {
	...withAddAction,
};
withRemoval.structure = structureWithCounterAndAction;

const withActions = {
	...withAddAction,
};
withActions.structure = structureWithActions;

const hugeStructure = [
	{
		name: 'Hitmonlee1',
		isOpened: true,
		children: [
			{
				name: 'Hitmonchan2',
				isOpened: true,
				children: [
					{
						name: 'Hitmonchan3',
						isOpened: true,
						children: [
							{
								name: 'Hitmonchan4',
								isOpened: true,
								children: [
									{
										name: 'Hitmonchan5',
										isOpened: true,
										children: [
											{
												name: 'Hitmonchan6',
												isOpened: true,
												children: [
													{
														name: 'Hitmonchan7',
														isOpened: true,
														children: [
															{
																name: 'Hitmonchan8',
																isOpened: true,
																children: [
																	{
																		name: 'Hitmonchan9',
																		isOpened: true,
																		children: [
																			{
																				name: 'Hitmonchan10',
																				isOpened: true,
																				children: [
																					{
																						name: 'Hitmonchen11',
																						isOpened: true,
																						children: [
																							{
																								name: 'Hitmonchen12',
																								isOpened: true,
																								children: [
																									{
																										name: 'Hitmonchen13',
																										isOpened: true,
																										children: [
																											{
																												name: 'Hitmonchen14',
																											},
																										],
																									},
																								],
																							},
																						],
																					},
																				],
																			},
																		],
																	},
																],
															},
														],
													},
												],
											},
										],
									},
								],
							},
						],
					},
				],
			},
		],
	},
];

const withDeepStructure = {
	...defaultProps,
};
withDeepStructure.structure = hugeStructure;

const cornerCaseLongName = {
	...defaultProps,
};
cornerCaseLongName.structure = [
	{
		name:
			'Hitmonlee1Hitmonlee1Hitmonlee1Hitmonlee1Hitmonlee1 Hitmonlee1Hitmonlee1Hitmonlee1Hitmonlee1Hitmonlee1',
		isOpened: true,
	},
];

const style = { width: '300px', border: '1px solid #eee', marginLeft: '10px' };

storiesOf('TreeView', module)
	.addDecorator(checkA11y)
	.addWithInfo('default', () => (
		<div>
			<h1>TreeView</h1>
			<h3>Definition</h3>
			<p>A view component to display any tree structure, like folders or categories.</p>
			<h3>Default property-set with action example: </h3>
			<div style={style}>
				<IconsProvider />
				<TreeView {...withAddAction} />
			</div>
		</div>
	))
	.addWithInfo('with custom icons', () => (
		<div>
			<h1>TreeView</h1>
			<h3>Definition</h3>
			<p>The icons can be customized, passign the Icon components props</p>
			<div style={style}>
				<IconsProvider />
				<TreeView {...withAddAction} structure={structureWithIcons} />
			</div>
		</div>
	))
	.addWithInfo('with custom header text', () => (
		<div>
			<h1>TreeView</h1>
			<h3>Definition</h3>
			<p>A view component to display any tree structure, like folders or categories.</p>
			<h3>Custom header and action tooltip property-set example: </h3>
			<div style={style}>
				<IconsProvider />
				<TreeView {...withHeader} />
			</div>
		</div>
	))
	.addWithInfo('without action', () => (
		<div>
			<h1>TreeView</h1>
			<h3>Definition</h3>
			<p>A view component to display any tree structure, like folders or categories.</p>
			<h3>Default property-set without action example: </h3>
			<div style={style}>
				<IconsProvider />
				<TreeView {...defaultProps} />
			</div>
		</div>
	))
	.addWithInfo('without header', () => (
		<div>
			<h1>TreeView</h1>
			<h3>Definition</h3>
			<p>A view component to display any tree structure, like folders or categories.</p>
			<h3>Default property-set without header example: </h3>
			<div style={style}>
				<IconsProvider />
				<TreeView {...defaultProps} noHeader />
			</div>
		</div>
	))
	.addWithInfo('with remove action and counter', () => (
		<div>
			<h1>TreeView</h1>
			<h3>Definition</h3>
			<p>A view component to display any tree structure, like folders or categories.</p>
			<h3>Default property-set with remove action example: </h3>
			<div style={style}>
				<IconsProvider />
				<TreeView {...withRemoval} />
			</div>
		</div>
	))
	.addWithInfo('with many actions', () => (
		<div>
			<h1>TreeView</h1>
			<h3>Definition</h3>
			<p>A view component to display any tree structure, like folders or categories.</p>
			<h3>Default property-set with remove action example: </h3>
			<div style={style}>
				<IconsProvider />
				<TreeView {...withActions} />
			</div>
		</div>
	))
	.addWithInfo('with deep structure', () => (
		<div>
			<h1>TreeView</h1>
			<h3>Definition</h3>
			<p>A view component to display any tree structure, like folders or categories.</p>
			<h3>Default property-set with deep structure: </h3>
			<div style={style}>
				<IconsProvider />
				<TreeView {...withDeepStructure} />
			</div>
		</div>
	))
	.addWithInfo('with long name', () => (
		<div>
			<h1>TreeView</h1>
			<h3>Definition</h3>
			<p>A view component to display any tree structure, like folders or categories.</p>
			<h3>Default property-set with cornercase: longname </h3>
			<div style={style}>
				<IconsProvider />
				<TreeView {...cornerCaseLongName} />
			</div>
		</div>
	));
