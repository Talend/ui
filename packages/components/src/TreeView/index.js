import TreeView from './TreeView.component';

export default TreeView;
