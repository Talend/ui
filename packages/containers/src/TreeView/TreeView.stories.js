import React from 'react';

import TreeView from '.';

export default {
	title: 'TreeView',
};

export const Default = () => (
	<TreeView
		id="my-treeview"
		collection="with.data"
		nameAttr="label"
		onToggleActionCreator="object:view"
		onSelectActionCreator="object:view"
		noHeader
	/>
);
