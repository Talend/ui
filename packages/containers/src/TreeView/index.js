import TreeView from './TreeView.container';

export default TreeView;
