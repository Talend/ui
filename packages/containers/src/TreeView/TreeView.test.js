import React from 'react';
import { shallow } from 'enzyme';
import mock from '@talend/react-cmf/lib/mock';
import Immutable from 'immutable';
import TreeView, {
	DEFAULT_STATE,
	DEFAULT_PROPS,
	transform,
	mapStateToProps,
} from './TreeView.container';

describe('TreeView', () => {
	let context;
	let state;
	let data;

	beforeEach(() => {
		context = mock.context();
		state = mock.state();
		data = new Immutable.List([
			new Immutable.Map({ id: 1, name: 'foo', children: [] }),
			new Immutable.Map({ id: 2, name: 'bar', children: [] }),
		]);
		context.store.getState = () => state;
	});

	it('should render the data', () => {
		const wrapper = shallow(<TreeView.WrappedComponent data={data} />, { context });
		expect(wrapper.getElement()).toMatchSnapshot();
	});

	it('should setState onClick', () => {
		const prevState = {
			state: DEFAULT_STATE,
		};
		const setState = jest.fn(fn => {
			prevState.state = fn(prevState);
		});
		const dispatchActionCreator = jest.fn();
		const onClick = jest.fn();
		const onClickActionCreator = 'my:action';
		const props = {
			setState,
			dispatchActionCreator,
			data,
			onClick,
			onClickActionCreator,
		};
		const wrapper = shallow(<TreeView.WrappedComponent {...props} />, { context });
		wrapper.simulate('click', data.get(0).toJS());
		expect(setState).toHaveBeenCalled();
		expect(prevState.state).not.toBe(DEFAULT_STATE);
		expect(prevState.state.get('opened').toJS()).toEqual([1]);
		expect(onClick).toHaveBeenCalledWith(data.get(0).toJS());
		expect(dispatchActionCreator).toHaveBeenCalled();
		expect(dispatchActionCreator.mock.calls[0][0]).toBe(onClickActionCreator);
		expect(dispatchActionCreator.mock.calls[0][1].props).toMatchObject(props);
		expect(dispatchActionCreator.mock.calls[0][2]).toEqual(data.get(0).toJS());
	});

	it('should close if re onClick', () => {
		const prevState = {
			state: DEFAULT_STATE,
		};
		const setState = jest.fn(fn => {
			prevState.state = fn(prevState);
		});
		const props = { setState, data };
		const wrapper = shallow(<TreeView.WrappedComponent {...props} />, { context });
		wrapper.simulate('click', data.get(0).toJS());
		expect(setState).toHaveBeenCalled();
		expect(prevState.state).not.toBe(DEFAULT_STATE);
		expect(prevState.state.get('opened').toJS()).toEqual([1]);
		wrapper.simulate('click', data.get(0).toJS());
		expect(setState.mock.calls.length).toBe(2);
		expect(prevState.state.get('opened').toJS()).toEqual([]);
	});

	it('should setState onSelect', () => {
		const prevState = {
			state: DEFAULT_STATE,
		};
		const setState = jest.fn(fn => {
			prevState.state = fn(prevState);
		});
		const dispatchActionCreator = jest.fn();
		const onSelect = jest.fn();
		const onSelectActionCreator = 'my:action';
		const props = {
			setState,
			dispatchActionCreator,
			data,
			onSelect,
			onSelectActionCreator,
		};
		const wrapper = shallow(<TreeView.WrappedComponent {...props} />, { context });
		wrapper.simulate('select', data.get(0).toJS());
		expect(setState).toHaveBeenCalled();
		expect(prevState.state).not.toBe(DEFAULT_STATE);
		expect(prevState.state.get('selectedId')).toEqual(1);
		expect(onSelect).toHaveBeenCalledWith(data.get(0).toJS());
		expect(dispatchActionCreator).toHaveBeenCalled();
		expect(dispatchActionCreator.mock.calls[0][0]).toBe(onSelectActionCreator);
		expect(dispatchActionCreator.mock.calls[0][1].props).toMatchObject(props);
		expect(dispatchActionCreator.mock.calls[0][2]).toEqual(data.get(0).toJS());
	});

	it('should unselect onSelect twice', () => {
		const prevState = {
			state: DEFAULT_STATE,
		};
		const setState = jest.fn(fn => {
			prevState.state = fn(prevState);
		});
		const onSelect = jest.fn();
		const props = { setState, data, onSelect };
		const wrapper = shallow(<TreeView.WrappedComponent {...props} />, { context });
		wrapper.simulate('select', data.get(0).toJS());
		expect(setState).toHaveBeenCalled();
		expect(prevState.state).not.toBe(DEFAULT_STATE);
		expect(prevState.state.get('selectedId')).toEqual(1);
		expect(onSelect).toHaveBeenCalledWith(data.get(0).toJS());
		wrapper.simulate('select', data.get(0).toJS());
		expect(prevState.state.get('selectedId')).toBe();
	});
});

describe('mapStateToProps', () => {
	it('should return props', () => {
		const state = mock.state();
		const data = new Immutable.Map({
			foo: new Immutable.Map({
				bar: new Immutable.List([new Immutable.Map({ foo: 'bar' })]),
			}),
		});
		state.cmf.collections = state.cmf.collections.set('data', data);
		const props = mapStateToProps(state, { collection: 'data.foo.bar' });
		expect(props.data).toBe(data.getIn(['foo', 'bar']));
	});
});

describe('transform', () => {
	it('should return undefined if no items', () => {
		expect(transform()).toBeUndefined();
	});

	it('add selected and toggled boolean every where', () => {
		const props = {
			...DEFAULT_PROPS,
			state: Immutable.Map({
				opened: Immutable.List([1, 11, 111]),
				selectedId: 11,
			}),
		};
		const items = [
			{
				id: 1,
				name: '1',
				children: [
					{
						id: 11,
						name: '11',
						children: [{ id: 111, name: '111' }],
					},
				],
			},
			{
				id: 2,
				name: '2',
			},
		];
		const structure = transform(items, props);
		expect(structure[0].id).toBe(1);
		expect(structure[0].selected).toBe(false);
		expect(structure[0].toggled).toBe(true);
		expect(structure[0].children[0].id).toBe(11);
		expect(structure[0].children[0].selected).toBe(true);
		expect(structure[0].children[0].toggled).toBe(true);
	});

	it("should unfold selected's parents", () => {
		const props = {
			...DEFAULT_PROPS,
			state: Immutable.Map({
				opened: Immutable.List([1, 11, 111]),
				selectedId: 111,
			}),
		};
		const items = [
			{
				id: 1,
				name: '1',
				children: [
					{
						id: 11,
						name: '11',
						children: [{ id: 111, name: '1111' }],
					},
				],
			},
			{
				id: 2,
				name: '2',
			},
		];

		const structure = transform(items, props);
		expect(structure[0].toggled).toBe(true);
		expect(structure[0].children[0].toggled).toBe(true);
		expect(structure[0].children[0].children[0].toggled).toBe(true);
		expect(structure[1].toggled).toBe(false);
	});
});
