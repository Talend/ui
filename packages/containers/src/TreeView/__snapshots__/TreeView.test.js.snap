// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeView should render the data 1`] = `
<TreeView
  addActionLabel="Add folder"
  childrenAttr="children"
  data={
    Immutable.List [
      Immutable.Map {
        id: 1,
        name: "foo",
        children: Array [],
      },
      Immutable.Map {
        id: 2,
        name: "bar",
        children: Array [],
      },
    ]
  }
  headerText="Folders"
  id="tc-treeview"
  idAttr="id"
  nameAttr="name"
  onSelect={[Function]}
  onToggle={[Function]}
  onToggleAllSiblings={[Function]}
  selectedId={undefined}
  structure={
    Array [
      Object {
        "children": Array [],
        "id": 1,
        "name": "foo",
        "parent": undefined,
        "toggled": false,
      },
      Object {
        "children": Array [],
        "id": 2,
        "name": "bar",
        "parent": undefined,
        "toggled": false,
      },
    ]
  }
/>
`;
